File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOBpMeasurementModel.kt
//
//  IDOBpMeasurementModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Blood pressure measurement event number

open class IDOBpMeasurementModel(status: Int, systolicBp: Int, diastolicBp: Int) : IDOBaseModel {

    /// 0: Not supported
    /// 1: Measurement in progress
    /// 2: Measurement successful
    /// 3: Measurement failed
    /// 4: Device in exercise mode 
    @SerializedName("status")
    var status: Int = status

    /// Systolic blood pressure
    @SerializedName("systolic_bp")
    var systolicBp: Int = systolicBp

    /// Diastolic blood pressure
    @SerializedName("diastolic_bp")
    var diastolicBp: Int = diastolicBp


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOBpMeasurementModel.swift
//
//  IDOBpMeasurementModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Blood pressure measurement event number
@objcMembers
public class IDOBpMeasurementModel: NSObject, IDOBaseModel {
    /// 0: Not supported
    /// 1: Measurement in progress
    /// 2: Measurement successful
    /// 3: Measurement failed
    /// 4: Device in exercise mode 
    public var status: Int
    /// Systolic blood pressure                                      
    public var systolicBp: Int
    /// Diastolic blood pressure                                     
    public var diastolicBp: Int
    
    enum CodingKeys: String, CodingKey {
        case status = "status"
        case systolicBp = "systolic_bp"
        case diastolicBp = "diastolic_bp"
    }
    
    public init(status: Int,systolicBp: Int,diastolicBp: Int) {
        self.status = status
        self.systolicBp = systolicBp
        self.diastolicBp = diastolicBp
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

